Engine/Translation/ITranslationProvider.cs

```diff
@@ -0,0 +1,173 @@
+//            Copyright Keysight Technologies 2012-2025
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace OpenTap.Translation;
+
+internal interface ITranslationProvider
+{
+    DisplayAttribute GetDisplayAttribute(IReflectionData mem);
+    DisplayAttribute GetDisplayAttribute(Enum e, string name);
+    string GetString(string key);
+}
+
+internal class ResXTranslationProvider : ITranslationProvider
+{
+    public CultureInfo Culture { get; }
+    private readonly Dictionary<string, DateTime> UpdateTime = [];
+    public ResXTranslationProvider(CultureInfo culture, IEnumerable<string> files)
+    {
+        Culture = culture;
+        foreach (var file in files)
+        {
+            // Mark all files as 
+            UpdateTime[file] = DateTime.MinValue;
+        }
+
+        TapThread.WithNewContext(() =>
```

Comment:
This class should probably be an IDisposable and clean up the thread when disposed.